zcash_primitives/src/transaction/components/orchard.rs

```diff
@@ -291,6 +342,53 @@ pub fn write_orchard_bundle<W: Write, D: OrchardDomainCommon + WriteBurn<W>>(
     Ok(())
 }
 
+/// Writes an [`orchard::Bundle`] in the appropriate transaction format.
+#[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+pub fn write_orchard_zsa_bundle<W: Write>(
+    mut writer: W,
+    bundle: Option<&orchard::Bundle<Authorized, Amount, OrchardZSA>>,
+) -> io::Result<()> {
+    if bundle.is_none() {
+        CompactSize::write(&mut writer, 0)?;
+        return Ok(());
+    }
+
+    let bundle = bundle.unwrap();
+
+    // We always have exactly one action group for orchardZSA
+    CompactSize::write(&mut writer, 1)?;
+
+    Vector::write_nonempty(&mut writer, bundle.actions(), |w, a| {
+        write_action_without_auth(w, a)
+    })?;
+
+    writer.write_all(&[bundle.flags().to_byte()])?;
+    writer.write_all(&bundle.anchor().to_bytes())?;
+    Vector::write(
+        &mut writer,
+        bundle.authorization().proof().as_ref(),
+        |w, b| w.write_u8(*b),
+    )?;
+
+    writer.write_u32::<LittleEndian>(0)?; // Currently timelimit is hardcoded to zero
```

Comment:
Please put the comment above the row to be consistent.
`timelimt must be zero for NU7`